cmd/analyze/main.go

```diff
@@ -142,6 +142,8 @@ func staticAnalysis(pkg *pkgecosystem.Pkg) {
 func main() {
 	log.Initialize(os.Getenv("LOGGER_ENV"))
 
+	flag.TextVar(&ecosystem, "ecosystem", pkgecosystem.Ecosystem(""), "ecosystem (npm, pypi, rubygems, packagist or crates.io)")
```

Comment:
I don't think this warrants moving into a function just to save a line of code.

In this case `flag.TextVar` is built specifically for this scenario. `pkgecosystem.Ecosystem` implements the `encoding.TextMarshaler` and `encoding.TextUnmarshaler` interface so it can be used in this fashion.

I think it is okay for `flag.Var` and `flag.TextVar` to be called in the main method prior to `flag.Parse()` (in fact, I've started thinking that global vars for flags aren't great either). 

FWIW I feel like `utils.CommaSeparatedFlags` could be improved as well:
 - it requires the use of the default flag.FlagSet, with no way to specify an alternative FlagSet.
 - it still requires an additional call to initialise (which is easy to forget).

What could help with reuse here is:
 1. adding a pkgecosystem.FlagHelp const string
 2. defining an empty ecosystem to avoid: `pkgecosystem.Ecosystem("")`.

Reply:
Good points. Regarding the global flags / main method placement, should we should be changing all the flags to be like this together rather than just having one different?

Reply:
I've made the reuse changes I mentioned.

Eventually I suggest putting all the flags into a separate function and returning something like a "config" struct.

But that can be done in a future PR.